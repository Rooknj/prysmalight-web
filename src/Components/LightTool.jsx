import React from "react";
import LightListQueryContainer from "./LightList/LightListQueryContainer";
import LightActionsContainer from "./LightActions/LightActionsContainer";

const LightTool = () => (
  <React.Fragment>
      <LightListQueryContainer />
      <LightActionsContainer />
  </React.Fragment>
);

export default LightTool;
