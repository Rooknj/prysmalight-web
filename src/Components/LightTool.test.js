import React from "react";
import { shallow, mount } from "enzyme";
import LightTool from "./LightTool";
import { createSerializer } from "enzyme-to-json";

// Mock out container components so that they will not get rendered while mounting
jest.mock("./LightList/LightListQueryContainer", () => () => (
  <div id="MockLightListQueryContainer">Mocked Light List Query Container</div>
));

// Add serializer to allow enzyme to create snapshots
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

it("renders without crashing", () => {
  shallow(<LightTool />);
});

it("matches shallow snapshot", () => {
  const shallowed = shallow(<LightTool />);
  expect(shallowed).toMatchSnapshot();
});

it("matches mounted snapshot", () => {
  const mounted = mount(<LightTool />);
  expect(mounted).toMatchSnapshot();
});
