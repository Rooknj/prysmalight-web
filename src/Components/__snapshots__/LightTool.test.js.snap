// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches mounted snapshot 1`] = `
Array [
  <div
    id="MockLightListQueryContainer"
>
    Mocked Light List Query Container
</div>,
  <div
    id="MockLightActionsContainer"
>
    Mocked Light Actions Container
</div>,
]
`;

exports[`matches shallow snapshot 1`] = `
<Fragment>
  <Component />
  <Component />
</Fragment>
`;
